plugins/modules/network_settings_workflow_manager.py

```diff
@@ -1043,167 +1318,316 @@ def get_network_params(self, site_id):
+                    "wired_data_collection": wired_data_collection,
+                    "wireless_telemetry": wireless_telemetry
+                }
+            }
+            network_settings = network_details.get("settings")
+
+            if snmp_details:
+                network_settings.update({"snmpServer": snmp_details})
+            else:
+                network_settings.update({"snmpServer": [""]})
+
+            if timezone_details is None:
+                network_settings.update({"timezone": {'identifier': 'GMT'}})
+            else:
+                network_settings.update({"timezone": timezone_details})
+
+            if syslog_details:
+                network_settings.update({"syslogServer": syslog_details})
+            else:
+                network_settings.update({"syslogServer": [""]})
+
+            if dhcp_details:
+                network_settings.update({"dhcpServer": dhcp_details})
+            else:
+                network_settings.update({"dhcpServer": [""]})
+
+            if dns_details is not None:
+                domain_name = dns_details.get("domainName")
+                if 'dnsServer' not in network_settings:
+                    network_settings['dnsServer'] = {}
+                if domain_name:
+                    network_settings.get("dnsServer").update({"domainName": dns_details.get("domainName")})
+                dns_servers = dns_details.get("dnsServers", [])
+                if len(dns_servers) > 0:
+                    network_settings.get("dnsServer").update({
+                        "primaryIpAddress": dns_details.get("dnsServers")[0]})
+                if len(dns_servers) > 1:
+                    network_settings.get("dnsServer").update({
+                        "secondaryIpAddress": dns_details.get("dnsServers")[1]})
             else:
                 network_settings.update({
-                    "clientAndEndpoint_aaa": {
-                        "network": aaa_pan_value,
-                        "protocol": aaa_value.get("protocol"),
-                        "ipAddress": aaa_value.get("ipAddress"),
-                        "servers": "ISE"
+                    "dnsServer": {
+                        "domainName": "",
+                        "primaryIpAddress": "",
+                        "secondaryIpAddress": ""
                     }
                 })
 
-        network_settings_snmp = network_settings.get("snmpServer")
-        if not network_settings_snmp.get("ipAddresses"):
-            network_settings_snmp.update({"ipAddresses": []})
+            if ntpserver_details is not None:
+                network_settings.update({"ntpServer": ntpserver_details})
+            else:
+                network_settings.update({"ntpServer": [""]})
+
+            if netflow_details is not None:
+                ip_address = netflow_details.get("collector").get("address")
+                port = netflow_details.get("collector").get("port")
+                if port is None:
```

Comment:
This check is not clear. If port is is None, then we are assigning a string. If not, we assign an integer. Can you please clarify?

Reply:
No need to check,  port is None, so removed this.